### PR TITLE
fixed vite reloading error

### LINKED ISSUE #102: Bug Report: Frontend Not Auto-Reloading in Docker Environment
When running the frontend application using Vite in a Docker container, changes in the code do not automatically trigger a page reload or reflect in the browser. 

https://github.com/user-attachments/assets/9fcb9a01-3e71-4806-a75e-6f33ce53a669

### PR DESCRIPTION
**Pull request by Yashgupta9330**
fixes #102 

https://github.com/user-attachments/assets/23a44612-9c7f-4de0-9ead-baf54a71f4f7

### CODE REVIEW

File: frontend/vite.config.ts
Comment (arkid15r) [inline frontend/vite.config.ts:1]: I'm not sure but it seems this is the only file that needs to be changed for getting it done.
Could you check this and revert all unrelated changes (if any)?

I'd also like you to search for a way to enable polling for [Windows machines only](https://vite.dev/config/server-options#server-watch) if possible. Let's narrow it down to affected environments only.
```

import os from 'os';

// Get the OS type
const osType = os.type();
```

File: frontend/Dockerfile.local
Comment (arkid15r) [inline frontend/Dockerfile.local:1]: I don't think we need to change this file as we run [npm commands here](https://github.com/OWASP/Nest/blob/e62de0696159ab6f858b64552c8fdfbfed51a46f/docker-compose.yaml#L56)

Comment (Yashgupta9330) [reply frontend/Dockerfile.local:1]: I have removed extra part in dockerfile.local

File: frontend/vite.config.ts [hunk 1]
```diff
@@ -1,7 +1,18 @@
 import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
+import os from 'os'
 
-// https://vite.dev/config/
+const platform = os.platform()
+console.log("platofrom" , platform)
```
Comment (arkid15r) [inline frontend/vite.config.ts]:
```suggestion
```

File: frontend/vite.config.ts [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
 export default defineConfig({
   plugins: [react()],
+  server: {
+    watch: {
+      usePolling: platform === 'win32' || platform === 'linux',
```
Comment (arkid15r) [inline frontend/vite.config.ts]:
```suggestion
      usePolling: os.platform() === 'win32',
```

Comment (arkid15r) [reply frontend/vite.config.ts]: We wanted it on Windows only, didn't we?

**Review by arkid15r — COMMENTED** [review]
A couple more suggestions:

File: frontend/vite.config.ts (hunk 2 above)
Comment (Yashgupta9330) [reply frontend/vite.config.ts]: I saw in google we need in linux also

Comment (arkid15r) [reply frontend/vite.config.ts]: I've made the changes based on https://vite.dev/config/server-options#server-watch

Could you provide your source?

Comment (Yashgupta9330) [reply frontend/vite.config.ts]: does the container os is linux?

Comment (Yashgupta9330) [reply frontend/vite.config.ts]: in console it is returning me linux

Comment (arkid15r) [reply frontend/vite.config.ts]: Let's focus on this: does hot reload work for you if you remove the 'linux' platform (keeping 'win32' only)?

Comment (Yashgupta9330) [reply frontend/vite.config.ts]: No

Comment (arkid15r) [reply frontend/vite.config.ts]: I see... As we run it in a container, the host OS is irrelevant here. Just submit it as always enabled (usePolling: true) and we'll take care of customization later when we have frontend env variables support.

Sorry for the hassle with this issue @Yashgupta9330
